esmvalcore/cmor/_fixes/icon/_base_fixes.py

```diff
@@ -0,0 +1,146 @@
+
+        Returns
+        -------
+        iris.cube.Cube
+            Desired cube.
+
+        Raises
+        ------
+        ValueError
+            Desired variable is not available in the input cubes.
+
+        """
+        if var_name is None:
+            var_name = self.extra_facets.get('raw_name',
+                                             self.vardef.short_name)
+        if not cubes.extract(NameConstraint(var_name=var_name)):
+            raise ValueError(
+                f"Variable '{var_name}' used to extract "
+                f"'{self.vardef.short_name}' is not available in input file")
+        return cubes.extract_cube(NameConstraint(var_name=var_name))
+
+    def get_horizontal_grid(self, cube):
+        """Get ICON horizontal grid from global attribute of cube.
+
+        Note
+        ----
+        If necessary, this functions downloads the grid file to a cache
+        directory. The downloaded file is valid for 7 days until it is
+        downloaded again.
+
+        Parameters
+        ----------
+        cube: iris.cube.Cube
+            Cube for which the ICON horizontal grid is retrieved. This cube
+            needs to have a global attribute that specifies the download
+            location of the ICON horizontal grid file (see
+            ``self.GRID_FILE_ATTR``).
+
+        Returns
+        -------
+        iris.cube.CubeList
+            ICON horizontal grid.
+
+        Raises
+        ------
+        ValueError
+            Input cube does not contain the necessary attribute that specifies
+            the download location of the ICON horizontal grid file (see
+            ``self.GRID_FILE_ATTR``).
+
+        """
+        if self.GRID_FILE_ATTR not in cube.attributes:
+            raise ValueError(
+                f"Cube does not contain the attribute '{self.GRID_FILE_ATTR}' "
+                f"necessary to download the ICON horizontal grid file:\n"
+                f"{cube}")
+        grid_url = cube.attributes[self.GRID_FILE_ATTR]
+
+        # If already loaded, return the horizontal grid (cube)
+        parsed_url = urlparse(grid_url)
+        grid_name = Path(parsed_url.path).name
+        if grid_name in self._horizontal_grids:
+            return self._horizontal_grids[grid_name]
+
+        # Check if grid file has recently been downloaded and load it if
+        # possible
+        grid_path = self.CACHE_DIR / grid_name
+        if grid_path.exists():
+            mtime = grid_path.stat().st_mtime
+            now = datetime.now().timestamp()
+            age = now - mtime
+            if age < self.CACHE_VALIDITY:
+                logger.debug("Using cached ICON grid file '%s'", grid_path)
+                self._horizontal_grids[grid_name] = self._load_cubes(grid_path)
+                return self._horizontal_grids[grid_name]
+            logger.debug("Existing cached ICON grid file '%s' is outdated",
+                         grid_path)
```

Comment:
there are a couple loose ends here - what if the first `if` doesn't get through? The second `if` throws only a debug message - anything you'd want to return instead maybe?

Reply:
I think that's fine to not use `else` statements here:

```python
l.94         if self.GRID_FILE_ATTR not in cube.attributes:
```
An error is raised immediately after the `if`, so no `else` necessary.

```python
l.104         if grid_name in self._horizontal_grids:
```
The function `return`s immediately after the `if`, so no `else` necessary.

```python
l.110         if grid_path.exists():
```
If the grid file does not exist, the function starts immediately downloading it (l.122).

```python
l.114             if age < self.CACHE_VALIDITY:
```
If the file age is younger, it is returned immediately after the `if`, so no `else` necessary. If it is older, the debug message is printed and the function continues to download the file again (l.122), so no return is necessary.

Reply:
aha, sounds good, wasn't too aware of what was happening there, just thought I'd flag it just in case :+1: